transformer_engine/pytorch/utils.py

```diff
@@ -179,8 +179,19 @@ def cast_if_needed(tensor: torch.Tensor, dtype: torch.dtype) -> torch.Tensor:
         return tensor if tensor is None or tensor.dtype == dtype else tensor.to(dtype)
 
 
-def check_dim_for_fp8_forward_exec(*tensors: Tuple[torch.Tensor, ...]) -> bool:
+def check_dim_for_fp8_forward_exec(*tensors: torch.Tensor) -> bool:
     """For fp8 fprop (TN layout), inputs and weights must be such
        that dim0 is divisible by 8 and dim1 is divisible by 16.
     """
     return all(not t.shape[0] % 8 and not t.shape[1] % 16 for t in tensors)
```

Comment:
```suggestion
def check_dim_for_fp8_forward_exec(tensor: torch.Tensor) -> bool:
    """For fp8 fprop (TN layout), inputs and weights must be such
       that dim0 is divisible by 8 and dim1 is divisible by 16.
    """
    return not tensor.shape[0] % 8 and not tensor.shape[1] % 16
```

Reply:
Right I think this change is fine since these are internal utilities not a part of the API, i.e. the only use of `check_dim_for_fp8_forward_exec` is in `assert_dim_for_fp8_forward_exec`.

Reply:
Committed!